### ISSUE TITLE
Don't auto-run applications until System Tray applet is loaded

### ISSUE THREAD

**Issue by Fjodor42** [linuxmint/cinnamon#3946]
Case in point is that I have the ownCloud application set to run upon login, but its icon never shows in the System Tray applet. Once in a while, it crashes, and when run again, the icon shows up.

This leads me to believe that applications that are set to run upon login may be executed before the System Tray applet is fully functional, which is sub-optimal.

On my work laptop, the same behaviour can be seen with the Dropbox client, leading me to believe that this is indeed a Cinnamon problem, as opposed to a problem with the respective applications.

I am far from sure, but there may be a correlation in time with the time that the startup animation was introduced.

**Comment by mtwebster** [comment]
Depending on how your app is set to autorun, you probably could just set a slight delay in **Startup Applications**.

Long term, we could look at having our session handler wait until the desktop is fully-formed to load these apps.

**Comment by tp0** [comment]
I noticed this recently with Dropbox. As a workaround setting the delay in the Startaup Application works.

**Comment by Fjodor42** [comment]
While I can confirm that setting a delay works around the problem, it would seem that at least Dropbox resets the delay to 0 when the client is updated.

For this reason, and since the default behaviour is suboptimal, I tend to think that a proper fix should be made - if at all possible.

**Comment by tp0** [comment]
@Fjodor42: Yes. Additionally, if you remove the auto-added Dropbox entry from the startup applications (and uncheck "start Dropbox on system startup" from Dropbox settings GUI) and add the same command there by yourself, it still doesn't seem to work with a delay or not. But I added these into a script loaded on boot and now it seems to load properly everytime:

```
sleep 10
dropbox start -i
```

**Comment by Vahan86** [comment]
@Fjodor42, is this still an issue for you with Cinnamon 3.2?

**Comment by MegaV0lt** [comment]
After two years still not fixed. Why?

**Comment by Fjodor42** [comment]
@Vahan86 I believe it is, yes - sorry

**Comment by Fjodor42** [comment]
Have any attempts actually been made to fix this?

**Comment by JosephMcc** [comment]
No, nor are there any real plans to since there is a built in way to get around this sort of thing.
